backend/cmm_helpers.ml

```diff
@@ -1246,6 +1246,40 @@ let get_field_computed imm_or_ptr mutability ~block ~index dbg =
   Cop
     (Cload { memory_chunk; mutability; is_atomic = false }, [field_address], dbg)
 
+(* Getters for unboxed int fields *)
+
+let get_field_unboxed_int32 mutability ~block ~index dbg =
+  let memory_chunk = Thirtytwo_signed in
+  (* CR layouts v5.1: We'll need to vary log2_size_addr to efficiently pack
+   * int32s *)
+  let field_address = array_indexing log2_size_addr block index dbg in
+  Cop
+    (Cload { memory_chunk; mutability; is_atomic = false }, [field_address], dbg)
+
+let get_field_unboxed_int64_or_nativeint mutability ~block ~index dbg =
+  let memory_chunk = Word_int in
+  let field_address = array_indexing log2_size_addr block index dbg in
+  Cop
+    (Cload { memory_chunk; mutability; is_atomic = false }, [field_address], dbg)
+
+(* Setters for unboxed int fields *)
+
+let setfield_unboxed_int32 arr ofs newval dbg =
+  (* CR layouts v5.1: We will need to vary log2_size_addr when int32 fields are
+     efficiently packed. *)
+  return_unit dbg
+    (Cop
+       ( Cstore (Thirtytwo_signed, Assignment),
+         [array_indexing log2_size_addr arr ofs dbg; newval],
+         dbg ))
```

Comment:
same remark as above the address of the load should probably be different on big-endian/little-endian architectures (or there should be an assertion+comment to state that we only care about one architecture)

Reply:
I've added an assertion. It shouldn't be hard to add support for big endian as-needed in the future.